scripts/upgrade

```diff
@@ -1,81 +1,156 @@
+if [ -z "$language" ]; then
+	language='en'
+	ynh_app_setting_set $app language $language
+fi
+
+#=================================================
+# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
+#=================================================
+
+# Backup the current version of the app
+ynh_backup_before_upgrade
+ynh_clean_setup () {
+	# restore it if the upgrade fails
+	ynh_restore_upgradebackup
+}
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
+#=================================================
+# CHECK THE PATH
+#=================================================
+
+# Normalize the URL path syntax
+path_url=$(ynh_normalize_url_path $path_url)
+
+#=================================================
+# STANDARD UPGRADE STEPS
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+
+# Download, check integrity, uncompress and patch the source from app.src
+ynh_setup_source "$final_path"
+
+#=================================================
+# NGINX CONFIGURATION
+#=================================================
+
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# CREATE DEDICATED USER
+#=================================================
+
+# Create a system user
 ynh_system_user_create $app
 
-# Modify dokuwiki conf
-sed -i "s@YNH_ADMIN_USER@$admin@g" ../conf/dokuwiki.php
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
+
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
 
-# Copy files to the right place
-final_path=/var/www/$app
-sudo mkdir -p $final_path
+# TODO Taken from old "upgrade" script. Should check if it is needed and what it does
+if [ -d "${final_path}/data/media" ]; then
+	chown -R $app:root $final_path/{data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp}
+fi
+
+# Set the "language" of Dokuwiki
+# Might overwrite user choice if language has been change directly in Dokuwiki admin panel
+# Should be enhance to avoid this. For now, it is easier to overwrite as there is no need to take care of existing configuration
+ynh_replace_string "__YNH_LANGUAGE__" "$language" "../conf/local.php"
 
-# Get source
-SETUP_SOURCE
+# Set the "admin" user of Dokuwiki
+ynh_replace_string "__YNH_ADMIN_USER__" "$admin" "../conf/local.php"
 
-sudo cp ../conf/dokuwiki.php $final_path/conf
+# Copy Yunohost specific configuration
+cp ../conf/local.php $final_path/conf
 
 # Do not override ACL configuration file
 if [ ! -f "$final_path/conf/acl.auth.php" ]; then
-	sudo cp ../conf/acl.auth.php $final_path/conf
+	cp ../conf/acl.auth.php $final_path/conf
 fi
 
 # Remove upgrade notification
 # See https://www.dokuwiki.org/update_check
```

Comment:
A short explanation of why could be great in addition to the link.

Reply:
Ok, it's an old commit.
Don't bother yourself with that, unless you know the reason of this line.

Reply:
Will add the comment:
> Check if a user configuration exists and do nothing if present.
> It will keep user customization's and settings instead of overwriting by the default configuration 

Maybe using `ynh_backup_if_checksum_is_different` and `ynh_store_file_checksum` is a better option combine with `if ... else`?

Could you elaborate a bit because the example is a little bit *too technical* and not *packager friendly* :)
https://github.com/YunoHost/example_ynh/blob/master/scripts/upgrade#L107-L110

Reply:
This helper works like this:
- During the install script, creating a config file.
- Using `ynh_store_file_checksum` to calculate a checksum of this file.
- During the upgrade script, using `ynh_backup_if_checksum_is_different` for comparing the checksum created during the install script with the checksum of the file right now.
- If the checksum is different, that means an admin has changed the file. So creating a backup of the previous config file.
- Then overwrite the file with the one from the package.

That's not perfect, because we overwrite the config file, but for now, we can't ask anything to users during the upgrade process.

Reply:
The problem with the second solution is always the same, in case of modification on our side, the file will not be fixed on the users part.
If, one day..., I find time to do it, I would like to write a "action" script which allow to restore any config file backup by this helper from the admin interface.
Still a real solution would be to ask to the user, or allow him, to overwrite or not his config files.
It could be an entry in the config panel.
But both of these solutions are currently only experimental and under test.

My point of view is this one,
- if the file could be modified from the app itself, from an admin panel or anything else, then we shouldn't overwrite the file during the upgrade.
- If the admin has to go modify the file by CLI on ssh, then we can overwrite it, because the admin shouldn't have modified the file like that, and he can restore the backup the same way if needed.